app/models/user.rb

```diff
@@ -289,15 +289,19 @@ def reset_api_key
   # only the assessment with the given id, if it is visible to the current user.
   # If it is not visible, returns an empty collection.
   def visible_assessments(assessment_type: nil, assessment_id: nil)
-    assessments = Assessment.where(is_hidden: false, type: assessment_type || Assessment.type)
+    assessments = Assessment.where(type: assessment_type || Assessment.type)
+    assessments = assessments.where(id: assessment_id) if assessment_id
     if self.section_id
-      assessments = Assessment.left_outer_joins(:assessment_section_properties)
```

Comment:
@mishaschwartz the root of the problem was this line. The previous version used `Assessment.left_outer_joins` and overwrite the query from the previous line, and so `assessment_type` was never used in this block of the code.

So the main change I made was to switch `Assessment.left_outer_joins` to `assessments.left_outer_joins`, so that in the new version both branches share the queries for `assessment_type` and `assessment_id`. (I could have left the `assessment_id` query below, but I thought it made a bit more sense to group the shared queries together. I don't care too strongly either way though.)